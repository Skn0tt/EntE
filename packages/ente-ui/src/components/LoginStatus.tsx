/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";

import {
  IconButton,
  ListItemText,
  ListItem,
  ListItemIcon
} from "@material-ui/core";
import { connect, Dispatch } from "react-redux";
import PowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import { logout, AppState, getDisplayname } from "../redux";
import { Action } from "redux";
import { Maybe } from "monet";

/**
 * # Component Types
 */
interface LoginStatusStateProps {
  displayname: Maybe<string>;
}
const mapStateToProps = (state: AppState): LoginStatusStateProps => ({
  displayname: getDisplayname(state)
});

interface LoginStatusDispatchProps {
  logout(): void;
}
const mapDispatchToProps = (
  dispatch: Dispatch<Action>
): LoginStatusDispatchProps => ({
  logout: () => dispatch(logout())
});

type LoginStatusProps = LoginStatusStateProps & LoginStatusDispatchProps;

/**
 * # Component
 */
export const LoginStatus: React.SFC<LoginStatusProps> = props => {
  const { displayname, logout } = props;

  return (
    <ListItem>
      <ListItemIcon>
        <IconButton aria-label="logout" onClick={logout} id="logout">
          <PowerSettingsNewIcon />
        </IconButton>
      </ListItemIcon>
      <ListItemText primary={displayname.orSome("")} />
    </ListItem>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginStatus);
