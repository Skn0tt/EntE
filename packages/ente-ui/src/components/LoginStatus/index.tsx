/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";

import {
  IconButton,
  ListItemText,
  ListItem,
  ListItemIcon
} from "@material-ui/core";
import { connect, Dispatch } from "react-redux";
import { PowerSettingsNew as PowerSettingsNewIcon } from "@material-ui/icons";
import { logout, AppState, getDisplayname } from "ente-redux";
import { Action } from "redux";

/**
 * # Component Types
 */
interface StateProps {
  displayname: string;
}
const mapStateToProps = (state: AppState): StateProps => ({
  displayname: getDisplayname(state)
});

interface DispatchProps {
  logout(): void;
}
const mapDispatchToProps = (dispatch: Dispatch<Action>): DispatchProps => ({
  logout: () => dispatch(logout())
});

type Props = StateProps & DispatchProps;

/**
 * # Component
 */
export const LoginStatus: React.SFC<Props> = props => {
  const { displayname, logout } = props;

  return (
    <ListItem>
      <ListItemIcon>
        <IconButton aria-label="logout" onClick={logout} id="logout">
          <PowerSettingsNewIcon />
        </IconButton>
      </ListItemIcon>
      <ListItemText primary={displayname} />
    </ListItem>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginStatus);
