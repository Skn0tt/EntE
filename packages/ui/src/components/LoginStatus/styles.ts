/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { StyleRules } from "material-ui/styles/withStyles";
import { Theme } from "material-ui/styles";

const styles = (theme: Theme): StyleRules => ({
  container: theme.mixins.toolbar
});

export default styles;
