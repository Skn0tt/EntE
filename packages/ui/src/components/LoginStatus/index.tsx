/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";

import styles from "./styles";
import { WithStyles } from "material-ui/styles/withStyles";
import { withStyles, IconButton, List } from "material-ui";
import { connect, Dispatch } from "react-redux";
import ListItem from "material-ui/List/ListItem";
import ListItemText from "material-ui/List/ListItemText";
import { PowerSettingsNew as PowerSettingsNewIcon } from "material-ui-icons";
import ListItemIcon from "material-ui/List/ListItemIcon";
import { logout, AppState, getDisplayname } from "ente-redux";
import { Action } from "redux";

/**
 * # Component Types
 */
interface StateProps {
  displayname: string;
}
const mapStateToProps = (state: AppState): StateProps => ({
  displayname: getDisplayname(state)
});

interface DispatchProps {
  logout(): void;
}
const mapDispatchToProps = (dispatch: Dispatch<Action>): DispatchProps => ({
  logout: () => dispatch(logout())
});

type Props = StateProps & DispatchProps & WithStyles;

/**
 * # Component
 */
export const LoginStatus: React.SFC<Props> = props => {
  const { classes, displayname, logout } = props;

  return (
    <div className={classes.container}>
      <List>
        <ListItem>
          <ListItemIcon>
            <IconButton aria-label="logout" onClick={logout} className="logout">
              <PowerSettingsNewIcon />
            </IconButton>
          </ListItemIcon>
          <ListItemText primary={displayname} />
        </ListItem>
      </List>
    </div>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(
  withStyles(styles)(LoginStatus)
);
