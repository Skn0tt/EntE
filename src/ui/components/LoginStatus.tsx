import * as React from "react";
import { ListItemText, ListItem, ListItemIcon } from "@material-ui/core";
import { useSelector } from "react-redux";
import SettingsIcon from "@material-ui/icons/Settings";
import { getDisplayname } from "../redux";
import Link from "next/link";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  container: {
    height: "64px",
  },
  icon: {
    margin: "12px",
  },
});

/**
 * # Component
 */
export const LoginStatus = (props: {}) => {
  const displayname = useSelector(getDisplayname);

  const classes = useStyles();

  return (
    <Link href="/preferences" aria-label="logout">
      <ListItem className={classes.container} button>
        <ListItemIcon className={classes.icon}>
          <SettingsIcon />
        </ListItemIcon>
        <ListItemText primary={displayname.orSome("")} />
      </ListItem>
    </Link>
  );
};

export default LoginStatus;
