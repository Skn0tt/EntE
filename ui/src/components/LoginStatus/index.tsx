import * as React from 'react';

import styles from './styles';
import { WithStyles } from 'material-ui/styles/withStyles';
import { withStyles, IconButton, List } from 'material-ui';
import { connect, Dispatch } from 'react-redux';
import * as select from '../../redux/selectors';
import { AppState } from '../../interfaces/index';
import ListItem from 'material-ui/List/ListItem';
import ListItemText from 'material-ui/List/ListItemText';
import { PowerSettingsNew as PowerSettingsNewIcon } from 'material-ui-icons';
import { Action } from 'redux-actions';
import { logout } from '../../redux/actions';
import ListItemIcon from 'material-ui/List/ListItemIcon';

interface StateProps {
  displayname: string;
}
const mapStateToProps = (state: AppState) => ({
  displayname: select.getDisplayname(state),
});

interface DispatchProps {
  logout(): Action<void>;
}
const mapDispatchToProps = (dispatch: Dispatch<Action<void>>) => ({
  logout: () => dispatch(logout()),
});

type Props = StateProps & DispatchProps & WithStyles;

const LoginStatus: React.SFC<Props> = props => (
  <div className={props.classes.container}>
    <List>
      <ListItem>
        <ListItemIcon>
          <IconButton aria-label="logout" onClick={() => props.logout()}>
            <PowerSettingsNewIcon />
          </IconButton>
        </ListItemIcon>
        <ListItemText primary={props.displayname} />
      </ListItem>
    </List>
  </div>
);

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(LoginStatus));
