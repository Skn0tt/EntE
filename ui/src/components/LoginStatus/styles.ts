import { StyleRules } from 'material-ui/styles/withStyles';
import { Theme } from 'material-ui/styles';

const styles = (theme: Theme): StyleRules => ({
  container: theme.mixins.toolbar,
});

export default styles;
