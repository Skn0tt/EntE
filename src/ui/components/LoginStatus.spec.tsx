/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { shallow } from "enzyme";
import { LoginStatus } from "./LoginStatus";
import { TestWrapper } from "test/TestWrapper";

describe("LoginStatus", () => {
  const comp = shallow(
    <TestWrapper>
      <LoginStatus />
    </TestWrapper>
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });
});
