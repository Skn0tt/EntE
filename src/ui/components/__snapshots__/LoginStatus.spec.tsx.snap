// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <LoginStatus />
</Provider>
`;
